demos/python_demos/speech_recognition_demo/ctcdecode-numpy/ctcdecode_numpy/word_prefix_set.cpp

```diff
@@ -29,15 +29,16 @@ size_t WordPrefixSet::add_words(const std::vector<std::vector<int> >& words) {
   std::sort(word_ptrs.begin(), word_ptrs.end(), lex_less);
 
   // Deduplicate
-  std::unique(word_ptrs.begin(), word_ptrs.end(), int_word_equal);
+  auto last_it = std::unique(word_ptrs.begin(), word_ptrs.end(), int_word_equal);
 
   trie_chars_.clear();
   trie_starts_.clear();
   trie_weights_.clear();
 
   // Go through the sorted list, and update children lists at every layer of trie.
   // The new child may only be the last child in each list.
-  for (auto word : word_ptrs) {
+  for (auto word_it = word_ptrs.begin(); word_it != last_it; word_it++) {
```

Comment:
This is insufficient; you also need to replace `word_ptrs.size()` with `last_it - word_ptrs.begin()` below.

(It might be simpler to do `word_ptrs.erase(last_it, word_ptrs.end());` instead.)